### DISCUSSION TITLE
Having mandatory and optional Columns

### DISCUSSION THREAD

**Discussion by bhelemilan** [ozlerhakan/poiji#203] [Ideas]
With **namedHeaderMandatory** option set to true, if we use **@ExcelCellName** annotation, the field is mapped to respective column in excel sheet and becomes a mandatory.

Can it be extended to have some optional columns? 

Example: An Excel sheet has 10 columns out of which 8 are mandatory and 2 to be optional. That means, we should be able to parse excel sheet with only 8 mandatory columns and with/without other 2 optional columns.

**Comment by ozlerhakan** [comment]
sounds a good idea @bhelemilan! , we can consider to have it.

**Comment by ozlerhakan** [comment]
with 3.1.2, we are able to use optional columns feature @bhelemilan
